.github/workflows/e2e.yml

```diff
@@ -24,9 +24,9 @@ jobs:
         rust: [ '1.65.0' ]
         # We build a dynamic-linked linux binary because otherwise HSM support fails with:
         #   Error: IO: Dynamic loading not supported
-        os: [ macos-11, ubuntu-20.04, ubuntu-22.04 ]
+        os: [ macos-12, ubuntu-20.04, ubuntu-22.04 ]
```

Comment:
This reminds me that https://github.com/dfinity/sdk/blob/master/.github/workflows/publish.yml#L35 should build a specific macos version.  Our e2e tests should be built against the same thing.